### PR TITLE
authz: remove v1alpha1 RBAC policy translation and e2e tests

### PR DESCRIPTION
**Pull request by yangminzhu**
This PR removes the v1alpha1 RBAC policy support. After this PR, applying the v1alpha1 RBAC policy will still be accepted by Istio but will be a no-op. Keeping the CRD definition and some internal data structure as it's needed by the `istioctl authz convert` command for automatically converting alpha to beta.

#22060

### CODE REVIEW

**Comment by yangminzhu** [comment]
> do we have some way to make it very obvious if a user uses the API that it does nothing? I feel like we should remove the CRD?

@howardjohn I will do some investigation about this, but after a second thought, it might be better to remove the CRD completely, we could either update the `istioctl auth convert` to support the conversion without CRD (maybe a little harder to so) or just remove the convert command as well.

**Comment by howardjohn** [comment]
@ostromart will we prune the CRD if we remove it?
I am not sure it's a good idea to actually remove it. Might make sense to
just keep it for now....

**Comment by yangminzhu** [comment]
@howardjohn Is this for upgrade? can we ask users to manually delete or convert their alpha policy before upgrading to 1.6? We have announced in 1.4 release that Istio will no longer support the alpha policy starting 1.6.

**Comment by howardjohn** [comment]
In 1.6 we support multiple control planes. So we cannot expect all alpha policies to be gone, since the old control plane may use them. We just can make the 1.6 one ignore them

File: pilot/pkg/security/authz/builder/builder.go [hunk 1]
```diff
@@ -136,9 +115,8 @@ func createTCPFilter(config *envoyRbacHttpPb.RBAC) *tcpFilterPb.Filter {
 	// The build function always return the config for HTTP filter, we need to extract the
 	// generated rules and set it in the config for TCP filter.
 	rbacConfig := &envoyRbacTcpPb.RBAC{
-		Rules:       config.Rules,
-		ShadowRules: config.ShadowRules,
```
Comment (liminw) [inline pilot/pkg/security/authz/builder/builder.go]: This PR is also trying to clean up all permissive mode policies? What is the current status of permissive mode policy? Is it completely broken already?

Comment (yangminzhu) [reply pilot/pkg/security/authz/builder/builder.go]: yes, permissive is not supported in beta policy, it should still work in alpha.

**Pull request merged**